plugin/src/main/java/io/jenkins/plugins/casc/impl/secrets/VaultSecretSource.java

```diff
@@ -164,6 +168,19 @@ private void readPropertiesFromVaultFile(String vaultFile, Properties prop) {
     @Override
     public Optional<String> reveal(String secret) {
         if (StringUtils.isBlank(secret)) return Optional.empty();
+
+        // TODO: move this to SecretSource.init() function which gets called only once when CasC.configure() is run
```

Comment:
@fishi0x01 did I merge it too soon? I can definitely implement the init call :smile:

Reply:
I have this implementation running since a few days now in our setup and there are no issues :+1: 
However, would be good to have a function that is called only once at `CasC.configure()` time, hence the `TODO`. Right now, each `reveal()` triggers a call to read ALL specified secret paths from vault. Those subsequent calls are unnecessary and should be avoided, but they do not necessarily do any harm. 

I can also have a look at this in the coming days :+1: